pkg/storage/db.go

```diff
@@ -712,8 +714,128 @@ func (db *DB) FetchByHostname(ctx context.Context, when time.Time, hostname stri
 
 // FetchByIP gets the assets who have IP address at the specified time
 func (db *DB) FetchByIP(ctx context.Context, when time.Time, ipAddress string) ([]domain.CloudAssetDetails, error) {
-	sqlstmt := fmt.Sprintf(latestStatusQuery, `aws_ips_ip`)
-	return db.runQuery(ctx, sqlstmt, ipAddress, when)
+	ver, err := db.GetSchemaVersion(ctx)
+	if err != nil {
+		return nil, err
+	}
+	var asset []domain.CloudAssetDetails
+	if ver < ReadsFromNewSchemaVersion {
+		sqlstmt := fmt.Sprintf(latestStatusQuery, `aws_ips_ip`)
+		asset, err = db.runQuery(ctx, sqlstmt, ipAddress, when)
+	} else {
+		ipaddr := net.ParseIP(ipAddress)
```

Comment:
ParseIP returns nil if the string passed to it is not a valid IP. Can you please add a check for this?

Reply:
Yes, I have added a check. Thank you for catching.